internal/net/grpc/stream.go

```diff
@@ -167,8 +167,8 @@ func BidirectionalStream[Q any, R any](
 }
 
 // BidirectionalStreamClient is gRPC client stream.
-func BidirectionalStreamClient(
-	stream ClientStream, dataProvider, newData func() any, f func(any, error),
+func BidirectionalStreamClient[S, R any](
+	stream ClientStream, sendDataProvider func() *S, callBack func(*R, error),
```

Comment:
@highpon Thank you for your contribution around important libs.

How about an interface that cancels all stream processing in case of false by using a boolean as the return value of the callback function argument of the BidirectionalStreamClient?

It would be very useful to have an interface that can stop all streams when a fatal error is detected in the callback function.

```suggestion
func BidirectionalStreamClient[S, R any](
	stream ClientStream, sendDataProvider func() *S, callBack func(*R, error) bool,
```